src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/vehicle_info_pending_resumption_handler.h

```diff
@@ -59,38 +59,67 @@ class VehicleInfoPendingResumptionHandler
       const smart_objects::SmartObject& response,
       const smart_objects::SmartObject& request) const;
 
-  bool IsResumptionResultSuccessful(
-      std::map<std::string, bool>& subscription_results);
-
-  void RemoveSucessfulSubscriptions(
-      std::set<std::string>& subscriptions,
-      std::set<std::string>& successful_subscriptions);
-
   smart_objects::SmartObjectSPtr CreateSubscribeRequestToHMI(
       const std::set<std::string>& subscriptions);
 
+  ns_smart_device_link::ns_smart_objects::SmartObject CreateFakeResponseFromHMI(
+      const std::map<std::string, smart_objects::SmartObject>& subscriptions,
+      const uint32_t fake_corrlation_id);
   void ClearPendingResumptionRequests() OVERRIDE;
 
  private:
-  void ClearPendingRequestsMap();
-
-  struct ResumptionAwaitingHandling {
-    const uint32_t app_id;
-    VehicleInfoAppExtension& ext;
-    resumption::Subscriber subscriber;
-
-    ResumptionAwaitingHandling(const uint32_t application_id,
-                               VehicleInfoAppExtension& extension,
-                               resumption::Subscriber subscriber_callback)
-        : app_id(application_id)
-        , ext(extension)
-        , subscriber(subscriber_callback) {}
+  void ProcessNextPendingResumption(
+      const smart_objects::SmartObject& response_message);
+  void TriggerPendingResumption();
+
+  struct PendingSubscriptionsResumption {
+    PendingSubscriptionsResumption(
+        const uint32_t app_id,
+        const uint32_t corr_id,
+        smart_objects::SmartObject request_message,
+        const std::set<std::string>& requested_vehicle_data)
+        : app_id_(app_id)
+        , fake_corr_id_(corr_id)
+        , requested_vehicle_data_(requested_vehicle_data) {}
+    uint32_t app_id_;
+    uint32_t fake_corr_id_;
+    std::set<std::string> requested_vehicle_data_;
+    std::set<std::string> restored_vehicle_data_;
+
+    std::map<std::string, smart_objects::SmartObject> subscription_results_;
+
+    bool Successfull() const;
```

Comment:
Not clear enough, this point is discussed below

Reply:
replied bellow